### DISCUSSION TITLE
Providing a mechanism to restart a PR workflow

### DISCUSSION THREAD

**Discussion by yarusome** [mdn/mdn-community#444] [Platform] [answered]
PR workflows can fail for unexpected external causes, e.g. [when `checkout` fails](https://github.com/mdn/content/actions/runs/6081807531/job/16498260666), but there is no way to restart the workflow by the contributors.

Note: Adding the manual trigger `workflow_dispatch` might enable abuse of resources, so it needs some other way.

**Comment by bsmth** [comment]
Thanks for raising this one. I think adding `workflow_dispatch` for check redirects might be okay, happy to hear other opinions. Would it be sufficient to [let us know in a chat](https://developer.mozilla.org/en-US/docs/MDN/Community/Communication_channels#discord_server) (Discord) for a low/no-tech solution to this? 

I've restarted the failed job in this case, anyway 😄

**Reply by yarusome** [reply]
Thanks for the restarting!

**Comment by yin1999** [comment, marked as answer]
Hey @yarusome. IMO, adding an extra mechanism to trigger the workflow might not be _necessary_.

The members with push permissions can rerun the failed workflows. And workflow failures due to downgrade of GitHub actions occur very rarely. Besides, the `workflow_dispatch` event could only triggered by member with `push` permission to the corresponding repo.

![image](https://github.com/mdn/mdn-community/assets/15844309/65f9068d-adf7-4fe1-a83e-39000052f8e0)

For the case where the workflow fails due to degradation of github action or other reasons, you could:

- ask for the reviewer to rerun the workflow
- force push or merge main branch (push a new commit) to retrigger the workflow

**Reply by bsmth** [reply]
>The members with push permissions can rerun the failed workflows.

I assumed this was the case; just to confirm @yarusome - do you have these UI elements available to restart jobs?

**Reply by yarusome** [reply]
@bsmth They're indeed not available on my UI. I'm fine with the status quo since it's only the third time that I've encountered this issue.

**Reply by yin1999** [reply]
Unfortunately, GitHub Action's performance does degrade yesterday and today. I'm not sure if this is related to the [release of action/checkout v4](https://github.com/actions/checkout/releases/tag/v4.0.0). Many people have reported [related issues](https://github.com/actions/checkout/issues?q=is%3Aissue+Failed+to+download+action+repository).

Hoping this issue to be resolved in the next few days.
